src/append.jl

```diff
@@ -0,0 +1,168 @@
+            error("fatal error writing arrow data")
+        end
+        @debug 1 "processing table partition i = $i"
+        tbl_schema = Tables.schema(tbl)
+
+        if !is_equivalent_schema(arrow_schema, tbl_schema)
+            throw(ArgumentError("Table schema does not match existing arrow file schema"))
+        end
+
+        if threaded
+            Threads.@spawn process_partition(tbl, dictencodings, largelists, compress, denseunions, dictencode, dictencodenested, maxdepth, msgs, alignment, i, sch, errorref, anyerror)
+        else
+            @async process_partition(tbl, dictencodings, largelists, compress, denseunions, dictencode, dictencodenested, maxdepth, msgs, alignment, i, sch, errorref, anyerror)
+        end
+    end
+    if anyerror[]
+        @error "error writing arrow data on partition = $(errorref[][3])" exception=(errorref[][1], errorref[][2])
+        error("fatal error writing arrow data")
+    end
+    # close our message-writing channel, no further put!-ing is allowed
+    close(msgs)
+    # now wait for our message-writing task to finish writing
+    wait(tsk)
+
+    b = FlatBuffers.Builder(1024)
+    schfoot = makeschema(b, sch[], firstcols)
+    if !isempty(blocks[1])
+        N = length(blocks[1])
+        Meta.footerStartRecordBatchesVector(b, N)
+        for blk in Iterators.reverse(blocks[1])
+            Meta.createBlock(b, blk.offset, blk.metaDataLength, blk.bodyLength)
+        end
+        recordbatches = FlatBuffers.endvector!(b, N)
+    else
+        recordbatches = FlatBuffers.UOffsetT(0)
+    end
+    if !isempty(blocks[2])
+        N = length(blocks[2])
+        Meta.footerStartDictionariesVector(b, N)
+        for blk in Iterators.reverse(blocks[2])
+            Meta.createBlock(b, blk.offset, blk.metaDataLength, blk.bodyLength)
+        end
+        dicts = FlatBuffers.endvector!(b, N)
+    else
+        dicts = FlatBuffers.UOffsetT(0)
+    end
+    Meta.footerStart(b)
+    Meta.footerAddVersion(b, Meta.MetadataVersion.V4)
+    Meta.footerAddSchema(b, schfoot)
+    Meta.footerAddDictionaries(b, dicts)
+    Meta.footerAddRecordBatches(b, recordbatches)
+    foot = Meta.footerEnd(b)
+    FlatBuffers.finish!(b, foot)
+    footer = FlatBuffers.finishedbytes(b)
+    Base.write(io, footer)
+    Base.write(io, Int32(length(footer)))
+    Base.write(io, "ARROW1")
+
+    return io
+end
+
+function footer_offset(bytes::Vector{UInt8})
+    magic_len = length(FILE_FORMAT_MAGIC_BYTES)
+    startoffset = length(bytes) - magic_len - sizeof(Int32) + 1
+    endoffset = length(bytes) - magic_len
+
+    read(IOBuffer(bytes[startoffset:endoffset]), Int32) + magic_len + sizeof(Int32)
+end
+
+function is_equivalent_schema(sch1::Tables.Schema, sch2::Tables.Schema)
+    (sch1.names == sch2.names) || (return false)
+    for (t1,t2) in zip(sch1.types, sch2.types)
+        (t1 <: Arrow.ArrowVector) && (t1 = eltype(t1))
```

Comment:
we shouldn't need this extra step after we fix the above code where we get tehs chema of the first table iterated by the stream

Reply:
done in https://github.com/JuliaData/Arrow.jl/pull/160/commits/6a56db3ad923e97a55f7131a86698a230ebcd9bd